sdk/storage/storage-blob/src/Clients.ts

```diff
@@ -4270,28 +4270,45 @@ export class BlockBlobClient extends BlobClient {
 
         const blockList: string[] = [];
         const blockIDPrefix = randomUUID();
-        let transferProgress: number = 0;
+        // Stores the amount of bytes progressed in each block
+        let transferProgressPerBlock: number[] = [];
 
         const batch = new Batch(options.concurrency);
         for (let i = 0; i < numBlocks; i++) {
+          // Initialize at 0
+          transferProgressPerBlock[i] = 0;
+          // Calculate block parameters
+          const start = blockSize * i;
+          const end = i === numBlocks - 1 ? size : start + blockSize;
+          const contentLength = end - start;
+          // Queue the block upload
           batch.addOperation(async (): Promise<any> => {
             const blockID = generateBlockID(blockIDPrefix, i);
-            const start = blockSize * i;
-            const end = i === numBlocks - 1 ? size : start + blockSize;
-            const contentLength = end - start;
             blockList.push(blockID);
             await this.stageBlock(blockID, bodyFactory(start, contentLength), contentLength, {
               abortSignal: options.abortSignal,
               conditions: options.conditions,
               encryptionScope: options.encryptionScope,
               tracingOptions: updatedOptions.tracingOptions,
+              onProgress(progress) {
```

Comment:
Looks good to me.  Though currently since `fetch` is used, we probably won't have real progress (i.e., still only getting one event after upload is finished.)

Reply:
@jeremymeng I thought that too, but @xirzec seems to think otherwise. You guys are definitely more familiar with the code here. https://github.com/Azure/azure-sdk-for-js/issues/32404#issuecomment-2584131700

Reply:
It is nice to enable it for the future, or potentially other http clients that support progress report.